### ISSUE TITLE
Openvidu Server docker image doesn't connect with coturn

### ISSUE THREAD

**Issue by miguelrcDEV** [OpenVidu/openvidu#184]
Hi again!
In my production machine I have:
- openvidu-server (2.7.0) released as a docker image.
- KMS 6.9.0 released as a docker image too.
- Coturn released as a service, installed since the Linux repo as you say in the docs.
- Redis as a service too.

The connections between openvidu-server, KMS and redis work correctly and manage to establish connections between them. The only problem is between the openvidu-server and the coturn:

![imagen_pegada_el_2019-01-28__2_18_pm](https://user-images.githubusercontent.com/17248343/51838769-9d298d00-2307-11e9-832d-97e644d76e22.png)

I have seen in this issue that it is closed (https://github.com/OpenVidu/openvidu/issues/83#issuecomment-401794008) that indeed the COTURN does not have support for docker images.

I want to continue using docker images since they give me versatility when it comes to updating version or doing a downgrade, so I would like to know what is the best way to deal with this problem? If it is better to create an image with the openvidu-server that contains the coturn (if you accept pull-request it would be just modify the dockerfile of the openvidu-server to install a coturn, I could do it if you want), or that another solution is it happens

Thanks in advance.

**Comment by pabloFuente** [comment]
We don't support COTURN integration in Docker setup yet because it is not a simple task to do so well. Adding a coturn and a redis database to the openvidu-server docker image is not the proper solution to go with Docker. Each one of those services should be in their own container, and everything should be run with docker-compose. But communications between openvidu-server and the turnadmin tool that manages TURN users is not direct, and must be performed through command-line on the COTURN host. That is feasible and relatively straightforward if openvidu-server and COTURN are running in the host machine without Docker, but is complicated when using containers.

To sum up, I cannot recommend to you an easy way to add COTURN support with the docker container of OpenVidu Server. You can try and build your own "monolithic" container with everything inside of it, but I can totatlly assure you that it's not the good way to go for a production environment.

**Comment by miguelrcDEV** [comment]
Hi Pablo!
In the end we have decided to launch openvidu-server from .jar. However, we have been trying to test the turn, but we have realized that something is missing because we have seen that in the api rest, specifically the POST a / api / token, in the "Sample return" the url that is Use as a token has 2 important parameters, I believe that the turn works correctly with the clients, which are turnUsername and turnCredential. In our case those 2 parameters do not reach us. In our case, the token request is made from a node server using the library you have for it (openvidu-node-client v2.5.0).

We have checked the logs of the openvidu-server to see that the coturn is configured correctly, and apparently it seems that if:

![captura de pantalla 2019-01-29 a las 13 14 41](https://user-images.githubusercontent.com/17248343/51908013-da098880-23c8-11e9-8d24-18fe54ce2707.png)

The parameters that we passed to the generateToken of the openvidu-node-client are something like this:

```
let tokenOptions = {
    session: sessionId,
    data: `{" userName ":" $ {userName} "," role ":" $ {role} "," participantId ":" $ {participantId} "},
    role: role
    kurentoOptions: {
         allowedFilters: ['FaceOverlayFilter', 'ChromaFilter', 'GStreamerFilter']
    }
}
```

What can we be doing wrong?
Thanks in advance

**Comment by pabloFuente** [comment]
Did you manage to solve this issue? If no TURN credentials are available inside the token, then it means that openvidu-server is not able to comunicate with the coturn server (via turnadmin command line tool).

**Comment by pabloFuente** [comment]
Since a few months have passed and some OpenVidu releases have been done (sorry for the delay), I'll close this issue for now. If there's still a problem with turn credentials, please open it again

**Comment by anidotnet** [comment]
I have been able to integrate coturn into openvidu docker image. Here is the docker image - 
`docker pull anidotnet/openvidu-server-kms-coturn:0.0.1`

I could not test it extensively, but I am able to get turn server details in generate token operation. If you verify the image, then I can create a PR to submit the changes.
